AZ-2402 improve profile list handling

smy-huddly opened the pull request:
When debugging the smarbase upgrade issue we have found an issue in LibUsbDotNet's caching of libusb's device list. When a device is plugged out from a port and a new device is plugged in to the same port before LibUsbDotNet re-enumerates the device list, it will assume it is the same device as it does not compare device parameters. We have found that a new device will get a new devicelist ID from libusb and here we add additional checks to see if this ID has changed. If so, we remove the original device from the cache and insert the new one 

Tracked in [AZ-2402](https://huddlyjira.atlassian.net/browse/AZ-2402)

thomas-huddly commented:
Do we need to manually bump the version, @eskilgh, like so?
https://github.com/Huddly/LibUsbDotNet/pull/1/files#diff-33f2f8743dd7164586785186733d5dfc03688d7c9be7cd173754b87161e852c1R3

eskilgh commented:
> Do we need to manually bump the version, @eskilgh, like so? https://github.com/Huddly/LibUsbDotNet/pull/1/files#diff-33f2f8743dd7164586785186733d5dfc03688d7c9be7cd173754b87161e852c1R3

Yes, that's how you would do it

thomas-huddly reviewed (approved):
👍 LGTM

thomas-huddly commented on stage/LibUsbDotNet/MonoLibUsb/Profile/MonoUsbProfileList.cs:
```diff
@@ -55,50 +55,52 @@ private void SetDiscovered(bool discovered)
+                }
+                else
+                {
+                    if (newProfile.ProfileHandle.DangerousGetHandle() != mList[iFoundOldIndex].ProfileHandle.DangerousGetHandle())
+                    {
+                        newProfile.mDiscovered = true;
+                        mList.Add(newProfile);
+                        FireAddRemove(newProfile, AddRemoveType.Added);
+                        mList[iFoundOldIndex].mDiscovered = false;
+                    }
+                    else
+                    {
+                        mList[iFoundOldIndex].mDiscovered = true;
+                        newProfile.mDiscovered = false;
+                    }
+                }
```
This block could be written using `else if` for readability and to reduce nesting

eskilgh replied:
Less  diff from the original version this  way though

smy-huddly replied:
Updated with continues for even better readability (in a C coder's eyes at least)

smy-huddly committed:
When debugging the smarbase upgrade issue we have found an issue in
LibUsbDotNet's caching of libusb's device list. When a device is plugged
out from a port and a new device is plugged in to the same
port before LibUsbDotNet re-enumerates the device list, it will assume
it is the same device as it does not compare device parameters. We have
found that a new device will get a new devicelist ID from libusb and
here we add additional checks to see if this ID has changed. If so,
we remove the original device from the cache and insert the new one

Tracked in AZ-2402

tmittet committed:
Bump version for Nuget release

thomas-huddly commented on stage/LibUsbDotNet/MonoLibUsb/Profile/MonoUsbProfileList.cs:
```diff
@@ -55,50 +55,52 @@ private void SetDiscovered(bool discovered)
+                    newProfile.mDiscovered = true;
+                    mList.Add(newProfile);
+                    FireAddRemove(newProfile, AddRemoveType.Added);
+                    continue;
```
Personally, I'm not a big fan of `continue` when used like this. I think `if`, `else if`, `else` would be clearer. But, there is no right or wrong here.

thomas-huddly reviewed (approved):
👍 LGTM
